File: Service/UserService.cs
using System.Collections.Generic;
using Domain;
using projectC.Repository;

namespace projectC.Service
{
    public class UserService
    {
        private readonly UserRepository userRepository;

        public UserService(UserRepository repository)
        {
            userRepository = repository;
        }

        public List<User> GetUsers() => userRepository.GetAll();

        public void AddUser(User user) => userRepository.Add(user);

        public bool AuthenticateUser(string username, string password)
        {
            var user = userRepository.GetById(username);
            return user != null && user.password == password;
        }

        public bool FindUser(string username) => userRepository.FindUser(username);

        public List<string> GetAllUsernames()
        {
            var result = new List<string>();
            foreach (var user in userRepository.GetAll())
            {
                result.Add(user.username);
            }
            return result;
        }
    }
}

File: project-java-BogdanBala112/HomeAssignment1/src/main/java/service/UserService.java
package service;

import Entities.User;
import Repository.UserRepository;

import java.util.ArrayList;
import java.util.List;

public class UserService {
    public UserRepository userRepository;

    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public List<User> getUsers(){
        return userRepository.getAll();
    }

    public void addUser(User user){
        userRepository.add(user);
    }

    public void removeUser(String username){
        userRepository.remove(username);
    }

    public void updateUser(String username, User user){
        userRepository.update(username, user);
    }

    public boolean authenticateUser(String username, String password){
        for (User user : userRepository.getAll()){
            if (user.getUsername().equals(username) && user.getPassword().equals(password))
                return true;
        }
        return false;
    }

    public boolean findUser(String username){
        return userRepository.findUserByUsername(username);
    }

    public List<String> getAllUsernames(){
        List<String> usernames = new ArrayList<>();

        for (User user : userRepository.getAll()){
            usernames.add(user.getUsername());
        }

        return usernames;
    }

}
